### ISSUE TITLE
twisted.conch.test.test_session constructs address objects with string port values

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5107]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#5107|
|Type|defect|
|Created|2011-05-15 12:41:01Z|
|Branch|https://github.com/twisted/twisted/tree/conch-int-ports-5107|

`return IPv4Address('TCP', 'remotehost', '8888')` and `return IPv4Address('TCP', 'localhost', '9999')` don't resemble real addresses an application might encounter.

<details><summary>Searchable metadata</summary>

```
trac-id__5107 5107
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_conch_int_ports_5107 branches/conch-int-ports-5107
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__tests tests
time__1305463261000000 1305463261000000
changetime__1305470153000000 1305470153000000
version__None None
owner__exarkun exarkun
cc__z3p
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [31858]) Branching to 'conch-int-ports-5107'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun removed owner|
|-|-|

Very simple, [build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/conch-int-ports-5107), please review.

**Comment by twisted-trac** [comment]
|[<img alt="jonathanj's avatar" src="https://avatars.githubusercontent.com/u/112166?s=50" width="50" height="50">](https://github.com/jonathanj)<a name="note_4"></a>|@jonathanj set owner to @exarkun|
|-|-|

Looks good. Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [31873]) Merge conch-int-ports-5107

Author: exarkun
Reviewer: jonathanj
Fixes: [#5107](https://github.com/twisted/twisted/issues/5107)

Construct `IPv4Address` objects in the Conch session tests using integer
port numbers instead of strings.
